misc/python/materialize/cli/ci_annotate_errors.py

```diff
@@ -167,16 +197,24 @@ def group_identical_errors(errors: list[str]) -> list[str]:
     return consolidated_errors
 
 
-def annotate_logged_errors(log_files: list[str]) -> int:
+def annotate_logged_errors(log_files: list[str], compare_against_main: bool) -> int:
     """
     Returns the number of unknown errors, 0 when all errors are known or there
     were no errors logged. This will be used to fail a test even if the test
     itself succeeded, as long as it had any unknown error logs.
     """
 
-    error_logs = get_error_logs(log_files)
+    errors = get_errors(log_files)
+
+    if not errors:
```

Comment:
Warrants a comment because it is not obvious on the first glance why we do this when `not errors`.

Reply:
I liked `error_logs` better than `error`; it is slightly more expressive. `errors_in_log` / `errors_from_log` even better. It makes it obvious that it is not a build error / build failure.

Reply:
It can be a build error now though since this PR also starts parsing the junit xml file. And `errors_from_logs_and_junit_xml_files` seems rather long.

Reply:
Does that mean that `errors` hold both log errors and test failures? If so, what further failure causes exist? If none, why do we need to compare against main?

Reply:
Sometimes a test fails but there is no information in any error log or junit xml.

Reply:
Ok. Please add that as comment